### DISCUSSION TITLE
【腾讯云代码分析】强化安全规则包

### DISCUSSION THREAD

**Discussion by cocorunning** [Tencent/CodeAnalysis#568] [Announcements]
![强化安全规则包首图](https://user-images.githubusercontent.com/91463108/186555696-bdd28ee7-0853-4e96-8209-a0d7406e344d.png)

该规则包重点关注各语言的**安全**类问题，针对OWASP Top10 中常见的漏洞进行分析，包括**注入攻击、 XML 外部实体攻击、XSS跨站脚本攻击、反序列化漏洞、敏感数据暴露、URL重定向漏洞**等，并结合CWE中常见漏洞，比如**服务端请求伪造（SSRF）**等，进行专项安全漏洞分析。
强化的安全规则主要有“致命”和“错误”级别，针对这些安全漏洞，TCA可以准确识别漏洞所在位置并提供修复建议。 

**已支持语言**：Go、PHP、Python
**已支持框架**：ThinkPHP、Django

注：该规则包由TCA独立工具支持，需申请授权免费使用，申请传送门：[《CLS使用文档》](https://tencent.github.io/CodeAnalysis/zh/quickStarted/enhanceDeploy.html#cls%E9%83%A8%E7%BD%B2)

### 启用规则包

分析方案 -> 代码检查 ->“强化”规则包 -> 启用/查看规则

![强化安全规则包](https://user-images.githubusercontent.com/91463108/186555881-af306ce5-f1da-42ba-87e8-eb58f3e2105e.png)

### 问题示例

反序列化漏洞
![反序列化漏洞](https://user-images.githubusercontent.com/91463108/186555985-4f49c7a9-d5be-45d7-aedd-069c3aa8411d.png)

XSS跨站脚本攻击
![XSS跨站脚本攻击](https://user-images.githubusercontent.com/91463108/186556012-0f18751e-024d-4952-9834-dc2eb6e36e07.png)

SSRF
![SSRF](https://user-images.githubusercontent.com/91463108/186556029-91b9260a-f95d-41c0-842b-45e286c5d82b.png)

命令行注入
![命令行注入](https://user-images.githubusercontent.com/91463108/186556058-ea18638d-0b15-4dbe-a9fb-d5a058d432ed.png)

SQL注入
![SQL注入](https://user-images.githubusercontent.com/91463108/186556076-1fe3a39c-345b-4824-bf86-7cc49edddeae.png)

### 公众号（腾讯云静态分析）回复「进群」获取官方微信交流群传送门_
![扫码_搜索联合传播样式-白色版](https://user-images.githubusercontent.com/91463108/182335753-173a8bfe-f38e-4e1e-9c81-c2df712d81b6.png)

**Comment by b3tterm4n** [comment]
希望不久后能强化一下java😄

**Comment by fangw1** [comment]
我们被veracode扫描到ssrf问题了,可是要如何解决呢 ？我们尝试使用ESAPI去修复， 但是没有效果。
```java
httpClient.execute(new HttpGet(url));
```
